### ISSUE TITLE
requested_oplock invalid value for 2.0.2 dialect

### ISSUE THREAD

**Issue by bulkans** [rapid7/ruby_smb#145]
in lib/ruby_smb/smb2/tree.rb line:94 create_request.requested_oplock = 0xff

requested_oplock is hardcoded to 0xff, but according to MSDN documentation this value is not valid for 2.0.2 dialect:

" SMB2_OPLOCK_LEVEL_LEASE   0xFF | A lease is requested. If set, the request packet MUST   contain an SMB2_CREATE_REQUEST_LEASE (section 2.2.13.2.8)   create context. This value is not valid for the SMB 2.0.2 dialect. "

https://msdn.microsoft.com/en-us/library/cc246502.aspx

This leads to error, when sending "request" to the server which complies with MSDN documentation.

**Comment by busterb** [comment]
What would you recommend changing it to? 0x8 or 0x1 ?

**Comment by cdelafuente-r7** [comment]
Absolutely, I think it is wrong to set `requested_oplock` to `0xFF` for any SMB dialect at this point. It requires a SMB2_CREATE_REQUEST_LEASE create-context structure and it is not implemented yet. Even if I've never seen any failure with Windows or Samba SMB servers, I would suggest to leave it to `0x00` for now.

Also, this comes from this commit: https://github.com/rapid7/ruby_smb/commit/9d41a675ecd115477949f3a1709433924f7ef535. Apparently, it is related to issues with very small files. This will need to be tested with that in mind.
